service_linux.go

```diff
@@ -167,7 +177,7 @@ func (c *controller) rmServiceBinding(name, sid, nid, eid string, vip net.IP, in
 	if len(s.loadBalancers) == 0 {
 		// All loadbalancers for the service removed. Time to
 		// remove the service itself.
-		delete(c.serviceBindings, sid)
+		delete(c.serviceBindings, skey)
```

Comment:
As discussed, should we protect this with a reference-counter to avoid removing this exposed port inadvertently when a service is active ?